src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -46,11 +47,21 @@ class PostgresDatabaseInfo : NpgsqlDatabaseInfo
         /// <inheritdoc />
         protected override IEnumerable<PostgresType> GetTypes() => _types ?? Enumerable.Empty<PostgresType>();
 
+        /// <summary>
+        /// The PostgreSQL version string as returned by the version() function. Populated during loading.
+        /// </summary>
+        public string LongVersion { get; set; } = default!;
+
         /// <summary>
         /// True if the backend is Amazon Redshift; otherwise, false.
         /// </summary>
         public bool IsRedshift { get; private set; }
 
+        /// <summary>
+        /// True if the backend is CockroachDb; otherwise, false.
+        /// </summary>
+        public bool IsCockroachDb { get; private set; }
```

Comment:
Personally I would like to avoid having such properties. Removing `IsRedshift` would be good too or at least hiding it since it used for `SupportsUnlisten`.

By the way, it would be cool if the long version contains information about Redshift. If so, we can safely drop the parameter from the connection string too.

Reply:
OK, I'll remove the property until a good reason comes along for us needing it. We don't actually need the long version either right now, but as I already did the work why not keep it, what do you think?

For Redshift, we have https://github.com/npgsql/npgsql/issues/2463 to track it. Feel free to do it :)

Reply:
Long version could be useful I think, so let's leave it.